src/TwigComponent/src/Resources/doc/index.rst

```diff
@@ -219,6 +219,50 @@ If an option name matches an argument name in ``mount()``, the option is
 passed as that argument and the component system will *not* try to set
 it directly on a property.
 
+ExposeInTemplate Attribute
+~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+.. versionadded:: 2.1
+
+    The ``ExposeInTemplate`` attribute was added in TwigComponents 2.1.
+
+All public component properties are available directly in your component
+template. You can use the ``ExposeInTemplate`` attribute to expose
+private/protected properties and public methods as variables directly in a
+component template (`someProp` vs `this.someProp`). For properties, they
+must be *accessible* (have a getter).
+
+    // ...
+    use Symfony\UX\TwigComponent\Attribute\ExposeInTemplate;
+
+    #[AsTwigComponent('alert')]
+    class AlertComponent
+    {
+        #[ExposeInTemplate]
+        private string $message; // available as `{{ message }}` in the template
+
+        #[ExposeInTemplate('alert_type')]
+        private string $type = 'success'; // available as `{{ alert_type }}` in the template
+
+        #[ExposeInTemplate]
+        public function getIcon(): string
+        {
+            // return value will be available as `{{ icon }}` in the template (note the get prefix is trimmed)
+        }
+
+        public function getMessage(): string
+        {
+            return $this->message;
+        }
+
+        public function getType(): string
+        {
+            return $this->type;
+        }
+
+        // ...
+    }
```

Comment:
Maybe a note about the methods:

```rst
.. note::

    When ``#[ExposeInTemplate]` is used above a method, that method will be called one
    time before the template is rendered.
```

Also... this is kind of calculated properties, isn't it? They're not "lazy" (they WILL be called), but I think it's as close as we're ever going to get (assuming some new feature isn't added to Twig)?

Reply:
Yeah, I don't know that I'd recommend using this for that (but this is probably because I really don't mind the `this.` syntax).

Reply:
The problem is that there IS no other computed prop syntax. If you call a method, it's called every time you use it. With this, the method IS called automatically (even if you never use it), but it's only called once, hence it is "computed".

Reply:
Yep, fair enough. I'm thinking of the scenario where you have a method that's a bit "heavy" and is only called under certain circumstances. Wouldn't want to expose this.